### ISSUE TITLE
Invalid function declarations in platform.windows interop module

### ISSUE THREAD

**Issue by dtretyakov** [JetBrains/kotlin-native#2727]
Currently for some functions in `platform.windows` module were generated signatures like that:

```kt
public val FormatMessage: kotlinx.cinterop.CPointer<kotlinx.cinterop.CFunction<(...) -> platform.windows.DWORD>>?
```

But it's expected that declaration should look like that:

```kt
public fun FormatMessage(...) -> platform.windows.DWORD
```

References:
* [FormatMessage function documentation](https://docs.microsoft.com/en-us/windows/desktop/api/winbase/nf-winbase-formatmessage)
* [FormatMessage function declaration in winbase.h file](https://github.com/tpn/winsdk-10/blob/master/Include/10.0.10240.0/um/WinBase.h#L2256-L2318).

Affected Kotlin native windows versions: 1.1.2, 1.3.11

**Comment by msink** [comment]
Use explicit `FormatMessageW` or `FormatMessageA`

**Comment by SvyatoslavScherbina** [comment]
> * [FormatMessage function declaration in winbase.h file](https://github.com/tpn/winsdk-10/blob/master/Include/10.0.10240.0/um/WinBase.h#L2256-L2318).

`FormatMessage` is declared this way:

```
#ifdef UNICODE
#define FormatMessage  FormatMessageW
#else
#define FormatMessage  FormatMessageA
#endif // !UNICODE
```

So current representation of this in Kotlin is not invalid, it is just straightforward, like
```
val FormatMessage = // pointer to FormatMessageW
```

**Comment by msink** [comment]
Really it's wrong - in C it is not pointer but alias, so in Kotlin it should be alias too.

**Comment by SvyatoslavScherbina** [comment]
Type of `FormatMessageW` itself is function type which is mostly equivalent to function pointer type.

**Comment by msink** [comment]
No - here `FormatMessageW()` is function type, but `FormatMessage` is alias to `FormatMessageW`
C preprocessor magic.

**Comment by SvyatoslavScherbina** [comment]
Kotlin/Native handles `FormatMessage` here like any other macro in most feasible straightforward way -- as a constant.
Moreover, C *language* doesn't have any kind of function aliases, and using preprocessor to achieve this is *convention*.
So the behaviour you mean actually requires supporting the convention, and I don't see how missing this support can be considered wrong behaviour. It is indeed inconvenient, but has nothing to do with *wrong*.

**Comment by msink** [comment]
Well, problem is that generated `kotlin/platform/windows/windows.kt` is 10MB (!!!) long, and handling this is very slow.

But:
- almost 2/3 here is garbage - `FormatMessage` and `FormatMessageA` probably never will be used
- MS documentation describes only `FormatMessage` as alias to `FormatMessageW`.

**Comment by SvyatoslavScherbina** [comment]
And now you are referring to other convention, but I'm not sure that it is valid. For example, `FormatMessageA` is present in both headers and documentation.

**Comment by msink** [comment]
Sure, but having options to enable such conventions would be nice.
Not high priority issue of course, but still.

**Comment by olonho** [comment]
Implementation-wise, it's by-product of macro-definition work implemented in 51f3123e9c7ef90aae006605aafb1b12cfea628d, and I do not perceive it as wrong, more like somewhat confusing in this case.

**Comment by olonho** [comment]
No further communications, closing.
